### PR TITLE
Fix for #18386 Use chart-utils to render x-Axis for StreamGraph

### PR DESCRIPTION
**Pull request by vtkalek**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,3 +2,15 @@
 [![Build Status](https://travis-ci.org/Microsoft/powerbi-visuals-streamgraph.svg?branch=master)](https://travis-ci.org/Microsoft/powerbi-visuals-streamgraph) [![Coverage Status](https://coveralls.io/repos/github/Microsoft/powerbi-visuals-streamgraph/badge.svg?branch=master)](https://coveralls.io/github/Microsoft/powerbi-visuals-streamgraph?branch=master)
 
 > A stacked area chart with smooth interpolation. Often used to display values over time.
+
+![stacked area chart screenshot](https://az158878.vo.msecnd.net/marketing/Partner_21474836617/Product_42949680594/Asset_1bbffb91-e55b-41b2-ad41-42b2698dfc6a/StreamGraphscreenshot1.png)
```
Comment (ignatvilesov) [inline README.md]: Why did you change this file? I suppose that you have to do `git rebase`

File: src/visual.ts [hunk 2]
```diff
@@ -310,15 +305,19 @@ module powerbi.extensibility.visual {
...
                         x: dataPointValueIndex,
                         y: isNaN(y)
                             ? StreamGraph.DefaultValue
                             : y,
                         text: label,
                         labelFontSize: fontSizeInPx
-                    });
+                    };
+                    series[valueIndex].dataPoints.push(streamDataPoint);
+                    if (streamDataPoint.x > xMaxValue) xMaxValue = streamDataPoint.x;
```
Comment (ignatvilesov) [inline src/visual.ts]: I would be better to move `xMaxValue = streamDataPoint.x` to the new line and use curly brackets around the statement.
Please consider it for the following statements.

File: src/visual.ts [hunk 3]
```diff
@@ -499,6 +504,238 @@ module powerbi.extensibility.visual {
...
             }
+
+            this.axes.attr("transform", SVGUtil.translate(this.margin.left, 0));
+            this.axisX.attr("transform", SVGUtil.translate(0, this.viewport.height - this.margin.bottom));
+            this.axisY.attr("transform", SVGUtil.translate(0, this.margin.top));
+
+            this.calculateAxes();
+            let xShow: boolean = this.data.settings.categoryAxis.show;
+            let yShow: boolean = this.data.settings.valueAxis.show;
+            if (xShow) {
```
Comment (ignatvilesov) [inline src/visual.ts]: These statements below are very similar. Can we encapsulate them into a function?

File: src/visual.ts [hunk 4]
```diff
@@ -499,6 +504,238 @@ module powerbi.extensibility.visual {
...
+        private static forcedTickSize: number = 150;
+        private static xLabelMaxWidth: number = 160;
+        private static xLabelTickSize: number = 3.2;
+        private calculateAxes() {
+            let xShow: boolean = this.data.settings.categoryAxis.show;
+            let yShow: boolean = this.data.settings.valueAxis.show;
+            let effectiveWidth: number = Math.max(0, this.viewport.width - this.margin.left - this.margin.right);
+            let effectiveHeight: number = Math.max(0, this.viewport.height - this.margin.top - this.margin.bottom);
+
+            let metaDataColumnPercent: powerbi.DataViewMetadataColumn = {
```
Comment (ignatvilesov) [inline src/visual.ts]: Why do we need this? Can we specify formatString instead?

File: src/visual.ts [hunk 5]
```diff
@@ -499,6 +504,238 @@ module powerbi.extensibility.visual {
...
+                type: ValueType.fromDescriptor({ numeric: true }),
+                objects: {
+                    general: {
+                        formatString: "0 %",
+                    }
+                }
+            };
+            if (xShow) {
+                let i: number, xDomain: number[] = [i = 0];
+                for (; i < this.data.xMaxValue; xDomain[i++] = i);
```
Comment (ignatvilesov) [inline src/visual.ts]: Can we use `d3.range` instead?

File: src/visual.ts [hunk 6]
```diff
@@ -499,6 +504,238 @@ module powerbi.extensibility.visual {
...
+                this.xAxisProperties = AxisHelper.createAxis({
+                    pixelSpan: effectiveWidth,
+                    dataDomain: xDomain,
+                    metaDataColumn: metaDataColumnPercent, // ??
```
Comment (ignatvilesov) [inline src/visual.ts]: Please remove the comment.

File: src/visual.ts [hunk 7]
```diff
@@ -499,6 +504,238 @@ module powerbi.extensibility.visual {
...
+                    metaDataColumn: metaDataColumnPercent,
+                    formatString: null,
+                    outerPadding: StreamGraph.outerPadding,
+                    isCategoryAxis: false,
+                    isScalar: true,
+                    isVertical: true,
+                    useTickIntervalForDisplayUnits: true
+                });
+            }
+
```
Comment (ignatvilesov) [inline src/visual.ts]: Please remove this empty line.

File: README.md (hunk 1 above)
Comment (vtkalek) [inline README.md]: Done

File: src/visual.ts [hunk 8]
```diff
@@ -511,15 +460,11 @@ module powerbi.extensibility.visual {
 
             this.dataView = options.dataViews[0];
 
-            if (options.type !== 4 && options.type !== 32 ) {
```
Comment (ignatvilesov) [inline src/visual.ts]: I suppose that you shouldn't get rid of this condition. It skips the converter if the update type is resize.

File: src/visual.ts [hunk 9]
```diff
@@ -568,6 +512,226 @@ module powerbi.extensibility.visual {
...
+        private static xLabelTickSize: number = 3.2;
+        private calculateAxes() {
+            let xShow: boolean = this.data.settings.categoryAxis.show;
+            let yShow: boolean = this.data.settings.valueAxis.show;
+            let effectiveWidth: number = Math.max(0, this.viewport.width - this.margin.left - this.margin.right);
+            let effectiveHeight: number = Math.max(0, this.viewport.height - this.margin.top - this.margin.bottom);
+
+            if (xShow) {
+                let i: number, xDomain: number[] = [i = 0];
+                for (; i < this.data.xMaxValue; xDomain[i++] = i);
```
Comment (ignatvilesov) [inline src/visual.ts]: Can we use d3.range instead?

File: src/visual.ts (hunk 4 above)
Comment (vtkalek) [inline src/visual.ts]: It's not working without it
